crates/dekoder/src/inventory.rs

```diff
@@ -0,0 +1,84 @@
+//! Assets manager.
+use glob::glob;
+use lz4_flex::frame::FrameDecoder;
+use ndarray::Array4;
+use ndarray_npy::NpzReader;
+use std::collections::HashMap;
+use std::ffi::OsString;
+use std::fs::{read_to_string, File};
+use std::io::{BufReader, Cursor};
+use std::path::PathBuf;
+use yaml_rust2::{Yaml, YamlLoader};
+
+use crate::{EKOError, Operator, Result};
+
+/// Headers are in yaml files.
+const HEADER_EXT: &str = "*.yaml";
+
+/// Assets manager.
+pub(crate) struct Inventory<K: Eq + for<'a> TryFrom<&'a Yaml, Error = EKOError>> {
+    /// Working directory
+    pub(crate) path: PathBuf,
+    /// Available keys
+    pub(crate) keys: HashMap<OsString, K>,
+}
+
+impl<K: Eq + for<'a> TryFrom<&'a Yaml, Error = EKOError>> Inventory<K> {
+    /// Load all available entries.
+    pub fn load_keys(&mut self) -> Result<()> {
+        let path = self.path.join(HEADER_EXT);
+        let path = path
+            .to_str()
+            .ok_or(EKOError::KeyError("due to invalid path".to_owned()))?;
+        for entry in glob(path)
```

Comment:
Since you're only using it here, and with a simple pattern, you could easily drop one dependency (though maintained by @rust_lang org as well), by using https://doc.rust-lang.org/std/fs/fn.read_dir.html and just filtering based on the extension.

I'm pretty sure you did this because it's exactly how it is written in Python. But in Python globbing is part of the standard library :P
(Python comes batteries-included, Rust tends to be minimal - even more minimal with `no_std`, but also the full standard library is not as extended as Python's one)

Reply:
done in c6784d9b